src/ralph/discovery/snmp.py

```diff
@@ -50,48 +57,59 @@ def user_data(auth, snmp_version, priv_protocol=cmdgen.usmDESPrivProtocol):
 
 
 def snmp_command(
-    hostname, community, oid, snmp_version='2c', timeout=1, attempts=3,
+    hostname, community, oid, snmp_version='2c', timeout=2, attempts=3,
     priv_protocol=cmdgen.usmDESPrivProtocol
 ):
     transport = cmdgen.UdpTransportTarget((hostname, 161), attempts, timeout)
-    data = user_data(community, snmp_version, priv_protocol=priv_protocol)
-    gen = cmdgen.CommandGenerator()
-    error, status, index, vars = gen.getCmd(data, transport, oid)
+    for priv_protocol in PRIV_PROTOCOLS:
+        data = user_data(community, snmp_version, priv_protocol=priv_protocol)
+        gen = cmdgen.CommandGenerator()
+        error, status, index, vars = gen.getCmd(data, transport, oid)
+        if not error:
+            return vars
```

Comment:
This `return` in loop is intentional? It's like "give me first nonerror result"?